src/query/service/src/pipelines/processors/transforms/group_by/aggregator_polymorphic_keys.rs

```diff
@@ -475,17 +474,50 @@ impl PolymorphicKeysHelper<HashMethodSerializer> for HashMethodSerializer {
 }
 
 #[derive(Clone)]
-pub struct PartitionedHashMethod<Method: HashMethod + Send> {
+pub struct PartitionedHashMethod<const BUCKETS_LG2: u32, Method: HashMethod + Send> {
     pub(crate) method: Method,
 }
 
-impl<Method: HashMethod + Send> PartitionedHashMethod<Method> {
-    pub fn create(method: Method) -> PartitionedHashMethod<Method> {
-        PartitionedHashMethod::<Method> { method }
+impl<const BUCKETS_LG2: u32, Method> PartitionedHashMethod<BUCKETS_LG2, Method>
+where
+    Self: HashMethod<HashKey = Method::HashKey>,
+    Method: HashMethod + PolymorphicKeysHelper<Method> + Send,
+    Method::HashKey: FastHash,
+{
+    pub fn create(method: Method) -> PartitionedHashMethod<BUCKETS_LG2, Method> {
+        PartitionedHashMethod::<BUCKETS_LG2, Method> { method }
+    }
+
+    pub fn create_partial_hashtable(
+        &self,
+    ) -> Result<PartitionedHashMap<Method::HashTable, BUCKETS_LG2, true>> {
+        let buckets = (1 << BUCKETS_LG2) as usize;
+        let mut tables = Vec::with_capacity(buckets);
+
+        for _index in 0..buckets {
+            tables.push(self.method.create_hash_table()?);
+        }
+
+        Ok(PartitionedHashMap::<_, BUCKETS_LG2, true>::create(tables))
+    }
+
+    pub fn create_final_hashtable(
+        &self,
+    ) -> Result<PartitionedHashMap<Method::HashTable, BUCKETS_LG2, false>> {
+        let buckets = (1 << BUCKETS_LG2) as usize;
```

Comment:
Final BUCKETS_LG2?

Reply:
`BUCKETS_LG2` is template here, it will be injected in construter